File: Haskell/FC5.hs
import Data.List
factors n = filter (\x -> rem n x == 0) [2..(floor (sqrt (fromIntegral n)))]
findFactor n = if null (factors n) then n else head (factors n) 

fc5 n = f [1] n
	where
		f factors 1 = sort factors
		f factors n = f (nub (concatMap (\x -> [x,x*factor]) factors)) (quot n factor)
			where factor = findFactor n

File: R/FC5.R
library(bit64)
# bit64 is a required package, to allow support for large numbers
# Large numbers must be placed in quotes to maintain precision!

options('scipen'=30)
findFactor = function(x) {
	r = 2:floor(sqrt(x))
	f = r[!x%%r][1]
	if (is.na(f)) return (x)
	else return (f)
}

fc5 = function(x) {
	n = as.integer64(x)
	factors = 1
	while (n>1) {
		f = findFactor(n)
		n = n/f
		factors = unique(c(factors,f*factors))
	}
	return(sort(factors))
}

#fc5("100000000000")

